.gitignore

```diff
@@ -1,5 +1,6 @@
 /target
 /crates/*/header-check/target
+/crates/*/header-check/Cargo.lock
```

Comment:
I renamed them originally because I wanted them to be part of the workspace, but that was untenable since the header check `build.rs` requires a path to illumos sources lest it fail loudly.  This made rust-analyzer and friends complain about the workspace containing said failing builds.

Considering how the header-check stuff is mostly out of the way for most repo consumers, I think it's probably fine?